commands/registrySettings.ts

```diff
@@ -1,36 +1,40 @@
+import * as assert from 'assert';
 import * as vscode from 'vscode';
 import { AzureRegistryNode } from "../explorer/models/azureRegistryNodes";
 import { CustomRegistryNode } from "../explorer/models/customRegistryNodes";
 import { DockerHubOrgNode } from "../explorer/models/dockerHubNodes";
 
+const defaultRegistryPathKey = "defaultRegistryPath";
+const defaultRegistryKey = "defaultRegistry";
+const combineRegistryPathsKey = "combineRegistryPathsFlag"
+
 export async function setRegistryAsDefault(node: CustomRegistryNode | AzureRegistryNode | DockerHubOrgNode): Promise<void> {
     let registryName: string;
     if (node instanceof DockerHubOrgNode) {
         registryName = node.namespace;
     } else if (node instanceof AzureRegistryNode) {
         registryName = node.registry.loginServer || node.label;
-    } else { //CustomREgistryNode
+    } else if (node instanceof CustomRegistryNode) {
         registryName = node.registryName;
+    } else {
+        assert.equal(false, true, "Type of registry node encountered not handled");
     }
 
     const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
-    await configOptions.update('defaultRegistryPath', registryName, vscode.ConfigurationTarget.Workspace);
-    vscode.window.showInformationMessage(`Update defaultRegistryPath to ${registryName}`);
+    await configOptions.update(defaultRegistryPathKey, registryName, vscode.ConfigurationTarget.Workspace);
+    vscode.window.showInformationMessage(`Updated defaultRegistryPath to ${registryName}`);
 }
 
 export async function consolidateDefaultRegistrySettings(): Promise<void> {
     const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
-    let defaultRegistryPath = configOptions.get('defaultRegistryPath', '');
-    let defaultRegistry = configOptions.get('defaultRegistry', '');
+    const combineRegistryPaths: boolean = configOptions.get(combineRegistryPathsKey);
+    let defaultRegistryPath: string = configOptions.get(defaultRegistryPathKey, '');
+    let defaultRegistry: string = configOptions.get(defaultRegistryKey, '');
```

Comment:
I don't think combineRegistryPathsFlag should be exposed to the user - I'd use ext.context.workspaceState for it.

Reply:
Then you can remove it from package.json.